lib/src/code_field/code_controller.dart

```diff
@@ -157,20 +164,47 @@ class CodeController extends TextEditingController {
     popupController = PopupController(onCompletionSelected: insertSelectedWord);
   }
 
-  void _analyzeCode() {
+  void _codeAnalysisCallback() {
     _debounce?.cancel();
-    _debounce = Timer(const Duration(milliseconds: 500), () async {
-      final text = _code.text;
-      final result = await analyzer.analyze(_code);
 
-      if (text != _code.text) {
-        return;
-      }
-      issues = result.issues;
-      notifyListeners();
+    if (analysisResult.analyzedCode.text == _code.text) {
+      return;
+    }
```

Comment:
Because we call notifyListeners after setting the issues to update the gutter widget, this method was calling itself 'recursively' because it is a subscription to the codeController.

Now it doesn't call analyzer if the code hasn't been changed since the last result